SelectRequestInput.jsx

```diff
@@ -1,59 +1,57 @@
+import { faCaretDown,faCaretUp } from "@fortawesome/free-solid-svg-icons";
 
-const SelectRequestInput = ({ handleChange, selected_value }) => {
-    const [link_hash_value, setLinkHashValue] = useState("");
-    const [hash_value_title, setHashValueTitle] = useState("");
-    const request_body = playground_requests.find(
-      el => (el.name || el.title) === selected_value
-    )
-    const default_value = request_body?.title || "Select API Call - Version 3"
-    playground_requests.sort((a, b) => a.title.localeCompare(b.title))
-
-    // To explain about the useEffects below:
-    // If the user clicks a link with a hash that redirects to the api-explorer page,
-    // the hash in the link will be used to immediately display the specific data.
-    useEffect(() => {
-        if (window.location.hash) {
-          const hash_value = window.location.hash.split("#")[1];
-          setLinkHashValue(hash_value);
-        }
-    }, [window.location.hash])
-    
-    useEffect(() => {
-      if (link_hash_value.length > 0) {
-        const find_select_value = playground_requests.find(el => el.name === link_hash_value);
-        setHashValueTitle(find_select_value?.title);
-      }
-    }, [link_hash_value, playground_requests])
-
-    // This useEffect has mainly a UX purpose, so that the user can see
-    // what function they selected in the link (hash).
-    useEffect(() => {
-      if (selected_value && request_body) {
-        window.location.hash = request_body ? request_body.name : window.location.hash || "";
-      }
-    }, [selected_value, request_body, window.location.hash])
+const SelectRequestInput = ({ selected, setSelected, handleChange }) => {
+  const [isActive,setIsActive] = useState(false)
+  const [toggle, setToggle] = useState(false)
+  library.add(faCaretDown);
+  library.add(faCaretUp);
+  const [searchResults, setSearchResults] = useState("");
 
     return (
-      <fieldset className={styles.apiRequest}>
-        <select
-          className={`${styles.select2} ${selected_value != "Select API Call - Version 3" ? styles.selected : ""}`}
-          onChange={handleChange}
-          defaultValue={link_hash_value ? hash_value_title : default_value}
-          id="settings-dropdown"
-        >
-          <option disabled className={styles.option}>{link_hash_value ? hash_value_title : default_value}</option>
-          <optgroup label="All calls">
-            {playground_requests.map(el => (
-              <option value={el.name} key={el.name}>
-                {el.title}
-              </option>
-            ))}
-          </optgroup>
-        </select>
-      </fieldset>   
+    <fieldset>
+      <div className={styles.dropdown}>
+      <div className= {styles.dropdownBtn} onClick={() => {
+        setIsActive(!isActive); 
+        setToggle(!toggle)
+      }}>
+        {selected}
+        {!isActive ? <FontAwesomeIcon icon="fa-solid fa-caret-down" /> : <FontAwesomeIcon icon="fa-solid fa-caret-up" /> }
```

Comment:
- For code readability purpose, better to check for `isActive` before `!isActive`.
- You should also add it as part of the next condition.